### PR TITLE
Alter the account and org_id columns of the hosts table.

### PR DESCRIPTION
**Pull request by roliveri**
# Overview

This PR is being created to address [ESSNTL-2528](https://issues.redhat.com/browse/ESSNTL-2528).

Alter the "account" column from nullable=False to nullable=True
Alter the "org_id" column from nullable=True to nullable=False

## PR Checklist

- [ ] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by thearifismail — COMMENTED** [review]
This look good but `lib/host_repository` and other supporting modules may need refactoring to require `org_id` and `account` as `optional`

**Comment by roliveri** [comment]
@thearifismail Yes. I believe Asa mentioned he would be working on those related changes after this PR is merged.

**Review by thearifismail — DISMISSED** [review]

**Review by kruai — CHANGES_REQUESTED** [review]
I am going to be changing a number of things in HBI after this is merged, but I think updating these fields on the DB models in `app/models.py` should be a part of this PR as well. Is that feasible?

**Comment by roliveri** [comment]
@kruai I'll look into that.
However, I can't figure out why the tests are failing. Maybe those changes are needed. All the tests pass when I run them locally.

File: tests/helpers/db_utils.py [hunk 1]
```diff
@@ -42,20 +42,24 @@ def _clean_tables():
 def minimal_db_host(**values):
     data = {
         "account": USER_IDENTITY["account_number"],
+        "org_id": USER_IDENTITY["account_number"],
```
Comment (kruai) [inline tests/helpers/db_utils.py:45]:
```suggestion
        "org_id": USER_IDENTITY["org_id"],
```

**Review by kruai — APPROVED** [review]
Never mind - I've made those changes in my PR, and changed things as necessary. They're all test data changes, so it doesn't matter to me if they go in here; I'll fix any remaining issues in my PR

**Pull request merged**